test/test_graph.jl

```diff
@@ -0,0 +1,70 @@
+## Test sparse
+# workspace()
+# using ProximalOperators
+# using Base.Test
+
+rng = MersenneTwister(1234)
+
+m, n = (10, 50)
+
+A = sprand( rng, Float64, m, n, 0.3)
+Adf = full(A)
+Ads = Adf';
+B = [A -speye(m)]
+Bd = full(B)
+
+fiag = ProximalOperators.IndGraph(A)
+fiaf = ProximalOperators.IndAffine(B, zeros(m))
+fiagdf = ProximalOperators.IndGraph(Adf)
+fiagds = ProximalOperators.IndGraph(Ads)
+
+c = randn(rng, Float64, n) * 10 + 10
+d = A * c
+
+v = randn(rng, Float64, m) * 10 + 10
+w = Ads * v
+
+@test fiag(c, d) == 0.0
+@test fiagdf(c, d) == 0.0
+@test fiagds(v, w) == 0.0
+
+y = zeros(d)
+x = zeros(c)
+
+xy = [x[:] ; y[:]]
+cd = [c[:] ; d[:]]
+
+x_a = @view xy[1:n]
+y_a = @view xy[n+1:end]
+
+prox!(x, y, fiag, c, d)
+@test fiag(x, y) == 0.0
+
+prox!(xy, fiaf, cd)
+@test fiaf(xy) == 0.0
+@test norm(x_a - x, Inf) < 1e-10
+@test norm(y_a - y, Inf) < 1e-10
+
+prox!(x, y, fiagdf, c, d)
+@test fiagdf(x, y) < 1e-10
+@test norm(x_a - x, Inf) < 1e-10
+@test norm(y_a - y, Inf) < 1e-10
+
+prox!(y, x, fiagds, v, w)
+@test fiagds(y, x) == 0.0
+
+@time for i = 1:10000
```

Comment:
These kind of tests are good to do on your own computer to make sure that you write efficient code, but we don't really need them in the package since we don't monitor the time in the tests, only the results. We have been thinking about writing some benchmark tests, but I don't think we have anything yet. Take a look at the other tests in the package to see how we are doing to test our other methods. I think all the code in this file could be moved into those automated tests.